cni/test/e2e-taint/e2e_taint_test.go

```diff
@@ -0,0 +1,389 @@
+			if err != nil {
+				return false
+			}
+			err = clientSet.CoreV1().Namespaces().Delete(context.TODO(), Test1NS.ObjectMeta.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				return false
+			}
+			err = clientSet.AppsV1().DaemonSets(Test2NS.Name).Delete(context.TODO(), Daemonset2Name, metav1.DeleteOptions{})
+			if err != nil {
+				return false
+			}
+			err = clientSet.CoreV1().Namespaces().Delete(context.TODO(), Test2NS.ObjectMeta.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				return false
+			}
+			return true
+		}, ShortTimeDuration, TimeInterval).Should(BeTrue())
+	})
+	It("[taint-controller features]  test on multiple labels in two namespace", func() {
+		TaintOptions = &taint.Options{
+			ConfigName:      "multi",
+			ConfigNameSpace: "default",
+		}
+		configmap, _ := buildConfigIfNotExists(clientSet, *TaintOptions, "testdata/multiConfig")
+		fmt.Println(configmap.Data)
+		taintSetter, err := taint.NewTaintSetter(clientSet, TaintOptions)
+		if err != nil {
+			log.Fatalf("Could not construct taint setter: %s", err)
+		}
+		addTaintToAllNodes(&taintSetter)
+		tc, err := taint.NewTaintSetterController(&taintSetter)
+		if err != nil {
+			log.Fatalf("Could not build controller %s", err)
+		}
+		stopCh := make(chan struct{})
+		go tc.Run(stopCh)
+		Consistently(func() bool {
+			nodeList, err := taintSetter.GetAllNodes()
+			if err != nil {
+				log.Fatalf("Could not fetch all nodes: %s", err)
+			}
+			for _, node := range nodeList.Items {
+				if !taintSetter.HasReadinessTaint(&node) {
+					return false
+				}
+			}
+			return true
+		}, ShortTimeDuration, TimeInterval).Should(BeTrue())
+		time.Sleep(ShortTimeDuration)
+		//create sample file
+		createDaemonset(&taintSetter.Client, "testdata/podsConfig/single-critical-label.yaml")
+		//it will not be ready because some labels in the namespace are not satisfied
+		Consistently(func() bool {
+			nodeList, err := taintSetter.GetAllNodes()
+			if err != nil {
+				log.Fatalf("Could not fetch all nodes: %s", err)
+			}
+			for _, node := range nodeList.Items {
+				if !taintSetter.HasReadinessTaint(&node) {
+					return false
+				}
+			}
+			return true
+		}, ShortTimeDuration, TimeInterval).Should(BeTrue())
+		time.Sleep(ShortTimeDuration)
+		createDaemonset(&taintSetter.Client, "testdata/podsConfig/addon-test2-in-test2.yaml")
+		//eventually all node's label should be removed because critical labels are satisfied
+		Eventually(func() bool {
+			nodes, _ := taintSetter.GetAllNodes()
+			for _, node := range nodes.Items {
+				if taintSetter.HasReadinessTaint(&node) {
+					return false
+				}
+			}
+			return true
+		}, LongTimeDuration, TimeInterval).Should(BeTrue())
+		close(stopCh)
+	})
+})
+var _ = Describe("[taint-controller features] single case", func() {
+	var clientSet client.Interface
+	var err error
+	var TaintOptions *taint.Options
+	BeforeEach(func() {
+		time.Sleep(6 * LongTimeDuration)
```

Comment:
Why do we need a sleep?

Reply:
sleep is for garbage collection purposes, if we use do not sleep at first, some resources may be deleted during other tests